spec/unit_spec.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'validate_environment_varables!' do
+  after(:each) { restore_env }
+
+  it 'exits if any environment variables are not set' do
+    Scraper.required_environment_variables.each { |var| unset_environment_variable(var) }
+    method = -> { Scraper.validate_environment_variables! }
+    expect(method).to raise_error(SystemExit)
+  end
+end
+
+describe 'posted_in_last_fortnight?' do
+  after(:each) { ScraperWiki.sqliteexecute('DELETE FROM data') }
+
+  it 'returns true if record exists' do
+    Scraper.record_successful_post('hello')
+    expect(Scraper.posted_in_last_fortnight?).to be true
+  end
+
+  it 'returns false if record does not exist' do
+    expect(Scraper.posted_in_last_fortnight?).to be false
+  end
+
+  it 'handles no database' do
+    # Create a new connection to new sqlite
+    ScraperWiki.close_sqlite
+    ScraperWiki.config = { db: Tempfile.new.path }
+    ScraperWiki.sqlite_magic_connection.execute('PRAGMA database_list')
+    # Test a query returns a false when there's no schema or data
```

Comment:
This comment points to using rspec `context` here to separate the setup/teardown from the expectation.

```
describe ".posted_in_last_fortnight?" do
  context 'when there is no schema or data' do
    before do
      # Create a new connection to new sqlite
      ScraperWiki.close_sqlite
      ScraperWiki.config = { db: Tempfile.new.path }
      ScraperWiki.sqlite_magic_connection.execute('PRAGMA database_list')
    end

    after do
      # Reset sqlite connection
      ScraperWiki.close_sqlite
      ScraperWiki.instance_variable_set(:@config, nil)
      ScraperWiki.instance_variable_set(:@sqlite_magic_connection, nil)
    end

    it { expect(Scraper.posted_in_last_fortnight?).to be false }
  end
end
```